c7n/resources/route53.py

```diff
@@ -485,3 +486,126 @@ def process(self, resources, event=None):
             elif not logging and not state:
                 results.append(r)
         return results
+
+
+@resources.register('resolver-logs')
+class ResolverQueryLogConfig(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'route53resolver'
+        arn_type = 'query_log_config'
+        enum_spec = ('list_resolver_query_log_configs', 'ResolverQueryLogConfigs', None)
+        name = 'Name'
+        id = 'Id'
+        cfn_type = 'AWS::Route53Resolver::ResolverQueryLoggingConfig'
+
+    permissions = (
+        'route53resolver:ListResolverQueryLogConfigs',
+        'route53resolver:ListResolverQueryLogConfigAssociations')
+
+    def augment(self, rqlcs):
+        client = local_session(self.session_factory).client('route53resolver')
+        for rqlc in rqlcs:
+            rqlc['Tags'] = self.retry(
+                client.list_tags_for_resource,
+                ResourceArn=rqlc['Arn'])['Tags']
+        return rqlcs
+
+
+@ResolverQueryLogConfig.action_registry.register('associate-vpc')
+class ResolverQueryLogConfigAssociate(BaseAction):
+    """Associates ResolverQueryLogConfig to a VPC
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: r53-resolver-query-log-config-associate
+            resource: resolver-logs
+            filters:
+              - type: value
+                key: 'Name'
+                op: eq
+                value: "Test-rqlc"
+            actions:
+              - type: associate-vpc
+                vpcid: all
+
+    """
+    permissions = (
+        'route53resolver:AssociateResolverQueryLogConfig',
+        'route53resolver:ListResolverQueryLogConfigAssociations')
+    schema = type_schema('associate-vpc', vpcid={'type': 'string',
+        'pattern': '^(?:vpc-[0-9a-f]{8,17}|all)$'}, required=['vpcid'])
+    RelatedResource = 'c7n.resources.vpc.Vpc'
+    RelatedIdsExpression = 'ResourceArn'
+
+    def get_vpc_id(self):
+        vpcs = RelatedResourceFilter.get_resource_manager(self).resources()
+        if self.data.get('vpcid') == 'all':
+            vpc_ids = [v['VpcId'] for v in vpcs]
+        else:
+            vpc_ids = [self.data.get('vpcid')]
+        return vpc_ids
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('route53resolver')
+        vpc_ids = self.get_vpc_id()
+
+        for r in resources:
+            for vpc_id in vpc_ids:
+                if not client.list_resolver_query_log_config_associations(Filters=[{'Name':
```

Comment:
I'm thinking this may lead to a bunch of repeat calls in cases where we're operating on multiple log configurations and/or multiple VPCs. It might be worth defining an annotation (like `c7n:Associations` maybe) as part of the `is-associated` filter you created below, and referring to that from here to avoid duplicate calls.

Reply:
@ajkerrigan many thanks for the suggestion ... got a quick question .. will annotation defined at the filter level still work  even if we are not using the filter in the policy with associate-vpc action?

Reply:
Good question @jasgrover , sorry I wasn't clear about that. No it wouldn't automatically work, you'd need to ensure that the annotation key was populated first. One pattern might be adding the annotation using something like a filter-level `augment` method. And then calling that filter-level augment from the action as needed.

The general approach would be similar to [this `s3-public-block` example](https://github.com/cloud-custodian/cloud-custodian/blob/efc0d1fd9aa4737b8a93cc0e0291643602a7077b/c7n/resources/account.py#L1518-L1529).

The related `set-s3-public-block` action [references the filter's annotation key](https://github.com/cloud-custodian/cloud-custodian/blob/efc0d1fd9aa4737b8a93cc0e0291643602a7077b/c7n/resources/account.py#L1592-L1593) to avoid making unnecessary calls. It still does [duplicate the `get_public_access_block` logic](https://github.com/cloud-custodian/cloud-custodian/blob/efc0d1fd9aa4737b8a93cc0e0291643602a7077b/c7n/resources/account.py#L1594-L1599), but it would be possible to reuse the filter's `augment` method there instead.

Reply:
@ajkerrigan .. made the changes you had suggested .. not sure if they are in line with your suggestions.

Reply:
I was thinking of doing it at the filter level rather than the resource level, but this will work too. Thanks :+1: . Just getting merge conflicts sorted, assuming checks pass we can merge this.